core/templates/dev/head/pages/library/SearchBarDirective.js

```diff
@@ -43,6 +43,81 @@ oppia.directive('searchBar', [
               }
             )
           );
+          $scope.activeMenuName = '';
+          $scope.ACTION_OPEN = 'open';
+          $scope.ACTION_CLOSE = 'close';
+          $scope.KEYBOARD_EVENT_TO_KEY_CODES = {
+            enter: {
+              shiftKeyIsPressed: false,
+              keyCode: 13
+            },
+            tab: {
+              shiftKeyIsPressed: false,
+              keyCode: 9
+            },
+            shiftTab: {
+              shiftKeyIsPressed: true,
+              keyCode: 9
+            }
+          };
+          /**
+           * Opens the submenu.
+           * @param {object} evt
+           * @param {String} menuName - name of menu, on which
+           * open/close action to be performed (category,language).
+           */
+          $scope.openSubmenu = function(evt, menuName) {
+            // Focus on the current target before opening its submenu.
+            angular.element(evt.currentTarget).focus();
+            $scope.activeMenuName = menuName;
+          };
+          $scope.blurNavigationLinks = function(evt) {
+            // This is required because if about submenu is in open state
+            // and when you hover on library, both will be highlighted,
+            // To avoid that, blur all the a's in nav, so that only one
+            // will be highlighted.
+            $('nav a').blur();
+          };
+          $scope.closeSubmenu = function(evt) {
+            $scope.activeMenuName = '';
+            angular.element(evt.currentTarget).closest('li')
+              .find('a').blur();
+          };
+          $scope.closeSubmenuIfNotMobile = function(evt) {
```

Comment:
Ditto here.

Reply:
I am sorry @bansalnitish , I was just trying to keep the code as consistent as possible

Reply:
Will surely remove the functions not being used :)

Reply:
No problem @geetchoudhary, just refer to the work done and look why this function was needed? Do we need it here too? Then take your decision as too how it should be handled.

Thanks!

Reply:
Sure, This was like my first PR. Thanks for the help.